context-monitoring/src/main/java/de/atb/context/persistence/monitoring/MonitoringDataRepository.java

```diff
@@ -146,6 +140,8 @@ protected synchronized void persist(final Model monitoringData, final Applicatio
             result.add((Type) initLazyModel(model, type));
         }
         lock.leaveCriticalSection();
+        set.commit();
```

Comment:
Shouldn't the `commit` happen before leaving the critical section?

Reply:
see other comment related to locking